core/server/models/api-key.js

```diff
@@ -50,6 +50,17 @@ const ApiKey = ghostBookshelf.Model.extend({
     refreshSecret(data, options) {
         const secret = createSecret();
         return this.edit(Object.assign({}, data, {secret}), options);
+    },
+
+    permittedOptions(methodName, options) {
```

Comment:
That is actually cool i think.

The destroy method does not need to do

options.require = true;

it automatically get's required assigned to it's options

Reply:
Yeah, so the caller will do something like `ApiKey.destroy({id: 1, require: true})`

Reply:
Don't we want to make that an auto-behaviour? 🤔We already do that for destroy actions in the model layer to get a proper 404. Let me know what you think :)

Reply:
I don't think want to make it auto, we could have internal calls that don't want errors being thrown?

Reply:
But if you don't use require, wouldn't you get a syntax error?
Because `obj` is not defined then?